### PR TITLE
test(spanner): troubleshoot backup timeouts

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #4306

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6979)

### CODE REVIEW

File: google/cloud/spanner/integration_tests/backup_integration_test.cc [hunk 1]
```diff
@@ -130,6 +137,11 @@ TEST_F(BackupTest, BackupTest) {
       << "Database " << db.database_id()
       << " not found in the backup operation list.";
 
+  // TODO(#4306) - CreateBackup() times out (rarely) log for troubleshoot
+  if (backup_future.wait_for(std::chrono::minutes(5)) != fs::ready) {
+    GCP_LOG(WARNING) << "Potential timeout in CreateBackup()" << db;
```
Comment (devbww) [inline google/cloud/spanner/integration_tests/backup_integration_test.cc:142]: I'm not sure what "potential timeout" messages are buying us.  Sometimes these operations just don't complete in the hour we've allotted.  But, OK.

Comment (devbww) [inline google/cloud/spanner/integration_tests/backup_integration_test.cc:142]: Nit: Could probably do with a space between `CreateBackup()` and the database name (et seq).

File: google/cloud/spanner/integration_tests/backup_integration_test.cc [hunk 2]
```diff
@@ -424,10 +448,15 @@ TEST_F(BackupTest, BackupTestWithCMEK) {
...
-          .get();
+  auto pending_create = database_admin_client_.CreateDatabase(
+      db, /*extra_statements=*/{}, encryption_config);
+
+  // TODO(#4306) - this test times out (rarely) additional logs to troubleshoot
+  if (pending_create.wait_for(std::chrono::minutes(5)) != fs::ready) {
+    GCP_LOG(WARNING) << "Potential timeout in CreateDatabase()" << in;
+    google::cloud::LogSink::Instance().Flush();
+  }
+  auto database = pending_create.get();
```
Comment (devbww) [inline google/cloud/spanner/integration_tests/backup_integration_test.cc:459]: Have we ever seen `CreateDatabase()` taking a long time?

**Review by devbww — APPROVED** [review]

File: google/cloud/spanner/integration_tests/backup_integration_test.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/spanner/integration_tests/backup_integration_test.cc:142]: I am not sure I follow.  Are you implying that these flakes are "normal" and therefore we should ignore them?

If so, I would be glad to change the tests to say something like "this operation is taking so long that we will simply declare success on this run and move on".  The downside if that these timeouts become "common", then we would not notice.

I am actually more interested in the output from the next line, where the last N RPCs would be logged. I am hoping those will help us identify if there was something particular about the instance / database / environment that triggered the timeout.

Comment (coryan) [reply google/cloud/spanner/integration_tests/backup_integration_test.cc]: Fixed.

File: google/cloud/spanner/integration_tests/backup_integration_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/spanner/integration_tests/backup_integration_test.cc]: Ooops, I changed my mind about what needed extra logging and did not revert this change. Fixed.

**Comment by coryan** [comment]
PTAL.  If you remain unconvinced by my rationale I will just close the PR.

File: google/cloud/spanner/integration_tests/backup_integration_test.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/spanner/integration_tests/backup_integration_test.cc]: Ack. Closing.

Comment (devbww) [reply google/cloud/spanner/integration_tests/backup_integration_test.cc]: It certainly might be nice to know we were actually in a `CreateBackup()` poll loop when the timeout happened, but I'm unaware of any hook we could use to dump the logging buffer when that happens.

**Pull request closed without merging**